UnknownHostException: Unable to resolve host "bsky.social.well-known"

Hi, I'm getting UnknownHostException on my Android app as soon as it tries to hit the well-known endpoint. 

- The app has the necessary Internet permission
- Tried on both physical device with working WiFi and 5G as well as on emulator

Full stack-trace:

```work.socialhub.kbsky.ATProtocolException
	at work.socialhub.kbsky.internal.share._InternalUtility.handleError(_InternalUtility.kt:120)
	at work.socialhub.kbsky.internal.share._InternalUtility.handleError$default(_InternalUtility.kt:99)
	at work.socialhub.kbsky.auth.internal._WellKnownResource.oAuthProtectedResource(_WellKnownResource.kt:68)
	at com.cyfertek.bsky_poster.AuthenticatorImp.authenticate(Authenticator.kt:43)
	at com.cyfertek.bsky_poster.MainViewModel$authenticate$1.invokeSuspend(MainViewModel.kt:29)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.internal.DispatchedContinuationKt.resumeCancellableWith(DispatchedContinuation.kt:375)
	at kotlinx.coroutines.intrinsics.CancellableKt.startCoroutineCancellable(Cancellable.kt:26)
	at kotlinx.coroutines.CoroutineStart.invoke(CoroutineStart.kt:358)
	at kotlinx.coroutines.AbstractCoroutine.start(AbstractCoroutine.kt:134)
	at kotlinx.coroutines.BuildersKt__Builders_commonKt.launch(Builders.common.kt:52)
	at kotlinx.coroutines.BuildersKt.launch(Unknown Source:1)
	at kotlinx.coroutines.BuildersKt__Builders_commonKt.launch$default(Builders.common.kt:43)
	at kotlinx.coroutines.BuildersKt.launch$default(Unknown Source:1)
	at com.cyfertek.bsky_poster.MainViewModel.authenticate(MainViewModel.kt:27)
	at com.cyfertek.bsky_poster.AppKt$App$1$2$1.invoke(App.kt:64)
	at com.cyfertek.bsky_poster.AppKt$App$1$2$1.invoke(App.kt:64)
	at com.cyfertek.bsky_poster.composables.LoginViewKt.LoginView$lambda$7$lambda$6$lambda$5(LoginView.kt:41)
	at com.cyfertek.bsky_poster.composables.LoginViewKt.$r8$lambda$ng3FS1rrQ9RlpEV-QlrmRNBolwY(Unknown Source:0)
	at com.cyfertek.bsky_poster.composables.LoginViewKt$$ExternalSyntheticLambda1.invoke(D8$$SyntheticClass:0)
	at androidx.compose.foundation.ClickableNode$clickPointerInput$3.invoke-k-4lQ0M(Clickable.kt:639)
	at androidx.compose.foundation.ClickableNode$clickPointerInput$3.invoke(Clickable.kt:633)
	at androidx.compose.foundation.gestures.TapGestureDetectorKt$detectTapAndPress$2$1.invokeSuspend(TapGestureDetector.kt:255)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTaskKt.resume(DispatchedTask.kt:163)
	at kotlinx.coroutines.DispatchedTaskKt.dispatch(DispatchedTask.kt:152)
	at kotlinx.coroutines.CancellableContinuationImpl.dispatchResume(CancellableContinuationImpl.kt:470)
	at kotlinx.coroutines.CancellableContinuationImpl.resumeImpl$kotlinx_coroutines_core(CancellableContinuationImpl.kt:504)
	at kotlinx.coroutines.CancellableContinuationImpl.resumeImpl$kotlinx_coroutines_core$default(CancellableContinuationImpl.kt:493)
	at kotlinx.coroutines.CancellableContinuationImpl.resumeWith(CancellableContinuationImpl.kt:359)
	at androidx.compose.ui.input.pointer.SuspendingPointerInputModifierNodeImpl$PointerEventHandlerCoroutine.offerPointerEvent(SuspendingPointerInputFilter.kt:719)
	at androidx.compose.ui.input.pointer.SuspendingPointerInputModifierNodeImpl.dispatchPointerEvent(SuspendingPointerInputFilter.kt:598)
	at androidx.compose.ui.input.pointer.SuspendingPointerInputModifierNodeImpl.onPointerEvent-H0pRuoY(SuspendingPointerInputFilter.kt:620)
	at androidx.compose.foundation.AbstractClickableNode.onPointerEvent-H0pRuoY(Clickable.kt:1044)
	at androidx.compose.ui.input.pointer.Node.dispatchMainEventPass(HitPathTracker.kt:387)
	at androidx.compose.ui.input.pointer.NodeParent.dispatchMainEventPass(HitPathTracker.kt:229)
	at androidx.compose.ui.input.pointer.HitPathTracker.dispatchChanges(HitPathTracker.kt:144)
	at androidx.compose.ui.input.pointer.PointerInputEventProcessor.process-BIzXfog(PointerInputEventProcessor.kt:120)
	at androidx.compose.ui.platform.AndroidComposeView.sendMotionEvent-8iAsVTc(AndroidComposeView.android.kt:1993)
	at androidx.compose.ui.platform.AndroidComposeView.handleMotionEvent-8iAsVTc(AndroidComposeView.android.kt:1944)
	at androidx.compose.ui.platform.AndroidComposeView.dispatchTouchEvent(AndroidComposeView.android.kt:1828)
	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3030)
	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2719)
	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3030)
	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2719)
	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3030)
	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2719)
	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3030)
	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2719)
	at com.android.internal.policy.DecorView.superDispatchTouchEvent(DecorView.java:440)
	at com.android.internal.policy.PhoneWindow.superDispatchTouchEvent(PhoneWindow.java:1830)
	at android.app.Activity.dispatchTouchEvent(Activity.java:3400)
	at com.android.internal.policy.DecorView.dispatchTouchEvent(DecorView.java:398)
	at android.view.View.dispatchPointerEvent(View.java:12752)
	at android.view.ViewRootImpl$ViewPostImeInputStage.processPointerEvent(ViewRootImpl.java:5106)
	at android.view.ViewRootImpl$ViewPostImeInputStage.onProcess(ViewRootImpl.java:4909)
	at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4426)
	at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:4479)
	at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:4445)
	at android.view.ViewRootImpl$AsyncInputStage.forward(ViewRootImpl.java:4585)
	at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:4453)
	at android.view.ViewRootImpl$AsyncInputStage.apply(ViewRootImpl.java:4642)
	at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4426)
	at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:4479)
	at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:4445)
	at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:4453)
	at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4426)
	at android.view.ViewRootImpl.deliverInputEvent(ViewRootImpl.java:7092)
	at android.view.ViewRootImpl.doProcessInputEvents(ViewRootImpl.java:7061)
	at android.view.ViewRootImpl.enqueueInputEvent(ViewRootImpl.java:7022)
	at android.view.ViewRootImpl$WindowInputEventReceiver.onInputEvent(ViewRootImpl.java:7195)
	at android.view.InputEventReceiver.dispatchInputEvent(InputEventReceiver.java:186)
	at android.os.MessageQueue.nativePollOnce(Native Method)
	at android.os.MessageQueue.next(MessageQueue.java:326)
	at android.os.Looper.loop(Looper.java:160)
	at android.app.ActivityThread.main(ActivityThread.java:6669)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
 
 Caused by: java.net.UnknownHostException: Unable to resolve host "bsky.social.well-known": No address associated with hostname
 	at java.net.Inet6AddressImpl.lookupHostByName(Inet6AddressImpl.java:157)
 	at java.net.Inet6AddressImpl.lookupAllHostAddr(Inet6AddressImpl.java:105)
 	at java.net.InetAddress.getAllByName(InetAddress.java:1154)
 	at okhttp3.Dns$Companion$DnsSystem.lookup(Dns.kt:49)
 	at okhttp3.internal.connection.RouteSelector.resetNextInetSocketAddress(RouteSelector.kt:169)
 	at okhttp3.internal.connection.RouteSelector.nextProxy(RouteSelector.kt:131)
 	at okhttp3.internal.connection.RouteSelector.next(RouteSelector.kt:73)
 	at okhttp3.internal.connection.ExchangeFinder.findConnection(ExchangeFinder.kt:205)
 	at okhttp3.internal.connection.ExchangeFinder.findHealthyConnection(ExchangeFinder.kt:106)
 	at okhttp3.internal.connection.ExchangeFinder.find(ExchangeFinder.kt:74)
 	at okhttp3.internal.connection.RealCall.initExchange$okhttp(RealCall.kt:255)
 	at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.kt:32)
 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
 	at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.kt:95)
 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
 	at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.kt:83)
 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
 	at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.kt:76)
 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
 	at okhttp3.internal.connection.RealCall.getResponseWithInterceptorChain$okhttp(RealCall.kt:201)
 	at okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:517)
 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
 	at java.lang.Thread.run(Thread.java:764)
 Caused by: android.system.GaiException: android_getaddrinfo failed: EAI_NODATA (No address associated with hostname)
 	at libcore.io.Linux.android_getaddrinfo(Native Method)
 	at libcore.io.BlockGuardOs.android_getaddrinfo(BlockGuardOs.java:172)
 	at java.net.Inet6AddressImpl.lookupHostByName(Inet6AddressImpl.java:137)
 	... 23 more```

--- a/auth/src/commonMain/kotlin/work/socialhub/kbsky/auth/internal/_WellKnownResource.kt
+++ b/auth/src/commonMain/kotlin/work/socialhub/kbsky/auth/internal/_WellKnownResource.kt
@@ -19,7 +19,7 @@ class _WellKnownResource(
         return proceed<WellKnownOAuthProtectedResourceResponse> {
             runBlocking {
                 HttpRequest()
-                    .url("${config.pdsServer}.well-known/oauth-protected-resource")
+                    .url("${config.pdsServer}/.well-known/oauth-protected-resource")
                     .accept(MediaType.JSON)
                     .get()
             }
@@ -37,7 +37,7 @@ class _WellKnownResource(
         return proceed<WellKnownOAuthAuthorizationServerResponse> {
             runBlocking {
                 HttpRequest()
-                    .url("${config.authorizationServer}.well-known/oauth-authorization-server")
+                    .url("${config.authorizationServer}/.well-known/oauth-authorization-server")
                     .accept(MediaType.JSON)
                     .get()
             }

--- a/auth/src/jvmTest/kotlin/work/socialhub/kbsky/auth/OAuthTest.kt
+++ b/auth/src/jvmTest/kotlin/work/socialhub/kbsky/auth/OAuthTest.kt
@@ -44,7 +44,7 @@ class OAuthTest : AbstractTest() {
     @Test
     fun tokenTest() {
         val callback = """
-            http://127.0.0.1/callback?iss=https%3A%2F%2Fbsky.social&state=7px3ldl0huvljddfqur00607mb57x9e55siwcpz3m00lrsx5i06geoyc65g06lmz&code=cod-bca898cc79bd729964492d72c356b37d1ed63561f23c67096844205d6ae56555
+            {{CALLBACK_URL}}
             """
             .trimIndent()
 

--- a/auth/src/jvmTest/kotlin/work/socialhub/kbsky/auth/WellKnownTest.kt
+++ b/auth/src/jvmTest/kotlin/work/socialhub/kbsky/auth/WellKnownTest.kt
@@ -1,6 +1,7 @@
 package work.socialhub.kbsky.auth
 
 import work.socialhub.kbsky.domain.Service.BSKY_SOCIAL
+import work.socialhub.kbsky.domain.Service.OYSTER_US_EAST
 import kotlin.test.Test
 
 class WellKnownTest : AbstractTest() {
@@ -8,14 +9,14 @@ class WellKnownTest : AbstractTest() {
     @Test
     fun testWellKnownOAuthProtectedResource() {
         val response = AuthFactory
-            .instance(BSKY_SOCIAL.uri)
+            .instance(OYSTER_US_EAST.uri)
             .wellKnown()
             .oAuthProtectedResource()
         println(response.data.authorizationServers[0])
     }
 
     @Test
-    fun testWell() {
+    fun testWellKnownOAuthAuthorizationServer() {
         val response = AuthFactory
             .instance(BSKY_SOCIAL.uri)
             .wellKnown()

